SU2_PY/OptimalPropeller.py

```diff
@@ -0,0 +1,371 @@
+
+    a = (w0*pow(Chi,2))/(pow(Chi,2)+pow((1+(w0)),2))
+    return a
+
+def Print_external_file(CTrs, CPrs):
+
+    """Function used to write the actuator disk input data file"""
+    file = open('ActuatorDisk.dat', 'w')
+    file.write('# Automatic generated actuator disk input data file using the Optimal Propeller code.\n')
+    file.write('# Data file needed for the actuator disk VARIABLE_LOAD type.\n')
+    file.write('# The load distribution is obtained using the inviscid theory of the optimal propeller\n')
+    file.write('# using global data.\n')
+    file.write('#\n')
+    file.write('# The first three lines must be filled.\n')
+    file.write('# An example of this file can be found in the TestCases directory.\n')
+    file.write('#\n')
+    file.write('# Author: Ettore Saetta, Lorenzo Russo, Renato Tognaccini.\n')
+    file.write('# Theoretical and Applied Aerodynamic Research Group (TAARG),\n')
+    file.write('# University of Naples Federico II\n')
+    file.write('# -------------------------------------------------------------------------------------\n')
+    file.write('#\n')
+    file.write('MARKER_ACTDISK= \n')
+    file.write('CENTER= \n')
+    file.write('AXIS= \n')
+    file.write('RADIUS= '+str(R)+'\n')
+    file.write('ADV_RATIO= '+str(J)+'\n')
+    file.write('NROW= '+str(Stations)+'\n')
+    file.write('# rs=r/R        dCT/drs       dCP/drs       dCR/drs\n')
+
+    for i in range(0, Stations):
+        file.write(f'  {r[i]:.7f}     {CTrs[i]:.7f}     {CPrs[i]:.7f}     0.0\n')
+
+    file.close()
+
+##########################
+###        Main        ###
+##########################
+# Screen output
+print('------------------ Optimal Propeller vsn 1.0.0 ------------------')
+print('| Computation of the optimal dCT/dr and dCP/dr distributions.   |')
+print('| Based on the inviscid theory of the optimal propeller.        |')
+print('|                                                               |')
+print('| This code is used to generate the actuator disk input data    |')
+print('| file needed for the VARIABLE_LOAD actuator disk type          |')
+print('| implemented in SU2 7.0.7.                                     |')
+print('|                                                               |')
+print('| Author: Ettore Saetta, Lorenzo Russo, Renato Tognaccini.      |')
+print('| Theoretical and Applied Aerodynamic Research Group (TAARG),   |')
+print('| University of Naples Federico II.                             |')
+print('-----------------------------------------------------------------')
+
+print('')
+print('Warning: present version requires input in SI units.')
+print('')
+
+# Number of radial stations in input.
+Stations = int(input('Number of radial stations: '))
+print('')
+
+dStations = float(Stations)
+
+# Resize the vectors using the number of radial stations.
+r = [None]*Stations
+chi = [None]*Stations
```

Comment:
Any reason why these variables are python lists and not numpy arrays?

Reply:
I am sorry, but it is the first time I code in Python. I have changed these variables and now they are numpy arrays.
Thanks for this comment!

Reply:
No problem, that was a genuine question, as maybe it was a requirement of some of the functions you are using.
But generally it is better to use numpy arrays for plain numbers since under the hood they are one block of memory.